### PR TITLE
Fixes strip rich formatting in the address bar

### PR DESCRIPTION
**Pull request by k0ur0x**
fixes https://github.com/mozilla-mobile/fenix/issues/1833

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt [hunk 1]
```diff
@@ -263,6 +265,29 @@ open class InlineAutocompleteEditText @JvmOverloads constructor(
...
 
+    private fun pasteAsPlainText(clipData: ClipData) {
+        var paste = ""
+        for (i in 0 until clipData.itemCount) {
+            val text = clipData.getItemAt(i).coerceToText(context)
+            paste += if (text is Spanned) text.toString() else text
+        }
+        val data = text.substring(0, selectionStart) + paste + text.substring(selectionEnd, text.length)
+        setText(data)
+    }
```
Comment (pocmo) [inline components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt]: We could consider moving this into `support-ktx` as an extension method, e.g. `ClipData.pasteAsPlainText()` (+ unit tests).

File: components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt [hunk 2]
```diff
@@ -263,6 +265,29 @@ open class InlineAutocompleteEditText @JvmOverloads constructor(
...
+        if (id == android.R.id.paste) {
+            return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                super.onTextContextMenuItem(android.R.id.pasteAsPlainText)
+            } else {
+                val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+                clipboard.primaryClip?.let { pasteAsPlainText(it) }
+                true
+            }
+        }
+        return super.onTextContextMenuItem(id)
```
Comment (pocmo) [inline components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt]: I wonder if we could cover this with a unit test?

Comment (k0ur0x) [reply components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt]: I added a unit test for that. Can you please review it to see if it is good or not?

File: components/support/ktx/src/test/java/mozilla/components/support/ktx/android/content/ClipDataTest.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        val view = EditText(context)
+        val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+        val spannableString = SpannableString("This is some bold text.").apply {
+            setSpan(StyleSpan(Typeface.BOLD), 0, length, 0)
+        }
+
+        clipboard.primaryClip = ClipData.newPlainText("bold text", spannableString)
+        clipboard.primaryClip?.pasteAsPlainText(view)
+        val spans = view.text.getSpans(0, view.length(), StyleSpan::class.java)
+        assertTrue(spans.isEmpty())
```
Comment (dector) [inline components/support/ktx/src/test/java/mozilla/components/support/ktx/android/content/ClipDataTest.kt]: I think we also need to check result text and have one more test where `EditText` has selection (as this helper deals with it as well).

File: components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt [hunk 4]
```diff
@@ -263,6 +265,19 @@ open class InlineAutocompleteEditText @JvmOverloads constructor(
         }
     }
 
+    override fun onTextContextMenuItem(id: Int): Boolean {
+        if (id == android.R.id.paste) {
```
Comment (dector) [inline components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt]: `<nerd mode>`

Single `return` statement is more preferable, I think. As it's more clear, less error-prone and [mentioned in Kotlin coding conventions](https://kotlinlang.org/docs/reference/coding-conventions.html#using-conditional-statements).

`</nerd mode>` :smile:

File: components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt (hunk 2 above)
Comment (dector) [reply components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt]: I suppose that it's nice to have tests for all cases here:
  - `android.R.id.paste` action on Post-M API;
  - `android.R.id.paste` action on Pre-M API;
  - not `android.R.id.paste` action.

File: components/ui/autocomplete/src/test/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditTextTest.kt [hunk 5]
```diff
@@ -334,4 +339,18 @@ class InlineAutocompleteEditTextTest {
...
+        val et = spy(InlineAutocompleteEditText(context, attributes))
+        val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+        val spannableString = SpannableString("This is some bold text.").apply {
+            setSpan(StyleSpan(Typeface.BOLD), 0, length, 0)
+        }
+
+        clipboard.primaryClip = ClipData.newPlainText("bold text", spannableString)
+        et.onTextContextMenuItem(android.R.id.paste)
+        val spans = et.text.getSpans(0, et.length(), StyleSpan::class.java)
+        assertTrue(spans.isEmpty())
```
Comment (dector) [inline components/ui/autocomplete/src/test/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditTextTest.kt]: And result text here as well.

**Pull request closed without merging**